### ISSUE TITLE
Fix Issues found using Go Vet

### ISSUE THREAD

**Issue by benyanke** [glauth/glauth#43]
A taste of the issues (dupes removed):

```
./configbackend.go:125: github.com/nmcclain/ldap.EntryAttribute composite literal uses unkeyed fields
./configbackend.go:132: github.com/nmcclain/ldap.Entry composite literal uses unkeyed fields
./configbackend.go:183: github.com/nmcclain/ldap.Entry composite literal uses unkeyed fields
./configbackend.go:188: github.com/nmcclain/ldap.ServerSearchResult composite literal uses unkeyed fields
./ldapbackend.go:64: return copies lock value: main.ldapHandler contains sync.Mutex
./ldapbackend.go:68: Bind passes lock by value: main.ldapHandler contains sync.Mutex
./ldapbackend.go:88: Search passes lock by value: main.ldapHandler contains sync.Mutex
./ldapbackend.go:128: Close passes lock by value: main.ldapHandler contains sync.Mutex
./ldapbackend.go:165: getSession passes lock by value: main.ldapHandler contains sync.Mutex
./ldapbackend.go:202: ping passes lock by value: main.ldapHandler contains sync.Mutex
./ldapbackend.go:245: getBestServer passes lock by value: main.ldapHandler contains sync.Mutex
``

**Comment by benyanke** [comment]
unkeyed fields fixed by PR #80

**Comment by icarus-sparry** [comment]
Whilst #80 is good to keep `go vet` quiet, the passing of locks by value are actual bugs in the code and need to be addressed. If you copy a lock structure and then lock the copy, the original is not locked!
